addons/mail/models/mail_link_preview.py

```diff
@@ -33,12 +35,15 @@ def _create_from_message_and_notify(self, message):
             return self
         urls = OrderedSet(html.fromstring(message.body).xpath('//a[not(@data-oe-model)]/@href'))
         link_previews = self.env['mail.link.preview']
+        ignore_patterns = self.env["ir.config_parameter"].sudo().get_param("mail.mail_link_preview_ignore_list", "")
         requests_session = requests.Session()
         link_preview_values = []
         link_previews_by_url = {
             preview.source_url: preview for preview in message.sudo().link_preview_ids
         }
         for url in urls:
+            if len(ignore_patterns) and any(re.compile(pattern).match(url) for pattern in ignore_patterns.split("\n")):
```

Comment:
It would be a little bit better to compile each regex only once outside of the loop.